### DISCUSSION TITLE
Serialising a schema object

### DISCUSSION THREAD

**Discussion by deepakguruu** [graphql-java/graphql-java#2284] [Q&A] [answered]
Are there any possible ways to serialise a schema object, instead of creating a new instance every time?.
In the case of a distributed n/w, this could be effective, as a schema can be serialised and stored in a distributed cache instead of creating a new instance in every server.

**Comment by dugenkui03** [comment, marked as answer]
[SchemaPrinter](https://github.com/graphql-java/graphql-java/blob/master/src/main/java/graphql/schema/idl/SchemaPrinter.java) can print `GraphQLSchema` to text. you can save `GraphQLSchema` in text, and create `GraphQLSchema` from text.

**Reply by deepakguruu** [reply]
SchemaPrinter can provide a textual representation of the Schema only right.
Is it possible to serialise all the relationships(data fetchers) in the schema as well?.

**Reply by dugenkui03** [reply]
`SchemaPrinter` can not serialise `DataFetcher` infomations.
